### ISSUE TITLE
ocean.util.log.Logger truncates in a way that Log did not

### ISSUE THREAD

**Issue by don-clugston-sociomantic** [sociomantic-tsunami/ocean#184]
Writing to a logger using ocean.util.log.Log ultimately does this:

```d
Logger format (Level level, cstring fmt, TypeInfo[] types, ArgList args)
        {
                if (types.length)
                {
                    if (buffer_ is null)
                        formatWithDefaultBuffer(level, fmt, types, args);
                    else
                        formatWithProvidedBuffer(level, fmt, types, args);
                }
                else
                   append (level, fmt);
                return this;
        }
```
Note that if no types are provided, it simply does an ``append`` rather than formatting the string.
``Logger`` behaves differently. It always formats. This means that it uses the buffer, which means the string is truncated at 2048 characters.

**Comment by don-clugston-sociomantic** [comment]
This is an easy issue to work around, it is possible to replace calls to ``log.error("xxx")`` to ``log.append(Level.Error, "xxx")``, which is arguably clearer; the main problem is that it is a silent difference in behaviour. This is not the only silent difference.

The more serious problem is that because of https://github.com/sociomantic-tsunami/ocean/issues/171, it does not appear possible to write logging code which produces the same output on D1 and D2.
Using ``Log``, typedefs are not printed in D2. Using ``Logger``, null strings are not printed correctly.

**Comment by mihails-strasuns-sociomantic** [comment]
@mathias-lang-sociomantic is it a bug or a feature? :)

**Comment by mathias-lang-sociomantic** [comment]
The previous behavior makes a lot of sense, and I definitely overlooked that case when changing that code, so definitely a bug. Working on a fix, but as usual, tests take longer to write than actual fix.

**Comment by mihails-strasuns-sociomantic** [comment]
Thanks!

**Comment by mathias-lang-sociomantic** [comment]
https://github.com/sociomantic-tsunami/ocean/pull/189

**Comment by nemanja-boric-sociomantic** [comment]
@mihails-strasuns-sociomantic @mathias-lang-sociomantic can this be moved to v3.2.5 milestone? I want to release v2.8.5 and this is the last issue., but it seems already fixed.

**Comment by mihails-strasuns-sociomantic** [comment]
Done

**Comment by mathias-lang-sociomantic** [comment]
Yes. Since Logger are only in v3 the milestone is wrong.

**Comment by nemanja-boric-sociomantic** [comment]
Fixed in #189
